include/libtorrent/session.hpp

```diff
@@ -131,6 +123,34 @@ namespace libtorrent
 		boost::shared_ptr<aux::session_impl> m_impl;
 	};
 
+	// The session_params is a parameters pack for configuring the session
+	// before it's started.
+	struct TORRENT_EXPORT session_params
+	{
+		// This constructor can be used to start with the default plugins
+		// (ut_metadata, ut_pex and smart_ban). The default values in the
+		// settings is to start the default features like upnp, nat-pmp,
+		// and dht for example.
+		session_params(settings_pack const sp = settings_pack());
```

Comment:
"const" should not be here. If the argument should be const, that's an implementation detail that should only be mentioned in the implementation. However, I suspect that we would like to allow this to be moved